src/core/player.js

```diff
@@ -151,7 +151,29 @@ DM.provide('Player',
         // see #5 : _domain.www should be protocol independent
         // remove protocol from existing value to preserve backward compatibility
         DM._domain.www = DM._domain.www.replace(/^https?\:/, '');
-        DM.Player._PROTOCOL = (window.location && /^https?:$/.test(window.location.protocol)) ? window.location.protocol : 'http:';
+
+        var location = null;
+
+        try {
+            try {
+              // try to get the top location first
+              location = window.top.location.href;
+            }
+            catch(e) {
+              // if not last ancestor
+              var ancestorOrigins = window.location.ancestorOrigins;
+              location = ancestorOrigins[ancestorOrigins.length - 1];
+            }
+        }
+        catch(e) {
+          // current window.location fallback
+          if (window.location) {
+              location = window.location.href;
+          }
+        }
+
+        var matchingProtocol = /^https?:/.exec(location);
+        DM.Player._PROTOCOL = matchingProtocol ? matchingProtocol[0] : 'https:';
```

Comment:
Wouldn't it be simpler to always use `https`, or there are any counter-indications?
@dharFr WDYT?

(one exception might be `file:` protocol but I don't think it's worth supporting it)

Reply:
at least to test locally, there is no serve yet.
You have to manually open this file in browser `file:///Dailymotion/dailymotion-sdk-js/tests/player.html`

Reply:
I guess we could maybe simplify. Worst case scenario is setting an https iframe within a parent `http` context.
Mixed content should be avoided in this case. Let's simplify ?

Reply:
Agree with @Cycymomo, the main use-case today is probably testing within a local environment (I used to use a simple local http server to test this during dev). We probably don't want to setup a certificate for a quick fix in the SDK, etc.